### PR TITLE
[REG-1950] Better manage event system lifecycle + fix legacy 2D mouse clicks

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Updates our code for managing our legacy input overrides... tries to keep these out of the way unless we're actively playing a replay
- Now handles detecting pointer over objects correctly based on our overrides
- Fixes an issue where 2D mouse over object hits weren't being detected properly

- This PR DOES NOT make the Match3 fully work with recording / replay... there are many more unique-ness-es about the game that require using behaviour segments instead of recording.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs [hunk 1]
```diff
@@ -320,64 +329,85 @@ public static void Update()
...
                     }
-                        
+
                     // 2D raycast
                     {
                         Vector3 mouseWorldPt = camera.ScreenToWorldPoint(new Vector3(_mousePosition.x, _mousePosition.y, camera.nearClipPlane));
-                        RaycastHit2D hit2D = Physics2D.Raycast(mouseWorldPt, Vector2.zero, distance: Mathf.Infinity, 
-                            layerMask: cameraRaycastMask);
-                        if (hit2D.collider != null)
+                        var collider = Physics2D.OverlapPoint(mouseWorldPt, cameraRaycastMask);
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs:376]: I'll make sure to do some extensive testing in my UI testing UAT to make sure this works as expected 👍

**Review by vontell — APPROVED** [review]
Thank you for these many bug fixes, good catch! As mentioned in our conversation in Slack, I'll make a ticket in the backlog for us to consider making a suite of automated game tests. I actually don't think it would be too difficult, just some busy work...

File: src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs:376]: Reminder, this only affects Legacy InputManager games

**Review by addisonbgross — APPROVED** [review]

File: src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs [hunk 2]
```diff
@@ -320,64 +329,85 @@ public static void Update()
...
                     {
                         continue;
                     }
-                        
-                    int cameraRaycastMask = camera.cullingMask & camera.eventMask;
-                        
+
+                    var cameraRaycastMask = camera.cullingMask & camera.eventMask;
+
+                    (GameObject, GameObject) result = (null, null);
```
Comment (batu) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs:362]: Flagging a github warning here

**Review by batu — APPROVED** [review]

File: src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs (hunk 2 above)
Comment (batu) [inline src/gg.regression.unity.bots/Runtime/Scripts/RGLegacyInputUtility/RGLegacyInputWrapper.cs:362]: flagging a `The variable 'result' is assigned but its value is never used` warning here.

**Pull request merged**